subprojects/base-services/src/main/java/org/gradle/internal/work/WorkerLeaseService.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.work;
+
+import org.gradle.internal.concurrent.Stoppable;
+import org.gradle.internal.resources.ProjectLeaseRegistry;
+
+public interface WorkerLeaseService extends WorkerLeaseRegistry, ProjectLeaseRegistry, Stoppable {
```

Comment:
What I've attempted to do is make `WorkerLeaseService` the main entry point that we would use in the vast majority of cases and `ResourceLockCoordinationService` would only be used if you need to do something more complicated that isn't captured by the convenience methods on `WorkerLeaseService`.  Of course, I'm sure this could be better and we'll improve it as we continue to use it.